src/components/VideoOptions.js

```diff
@@ -1,31 +1,70 @@
 import "../styles/components/videoOptions.css";
-import { usePlaylist } from "../context/playlist-context";
+import { useVideoAction } from "../context/video-action-context";
 import { RequiresAuth } from "../routes/RequiresAuth";
 import { useAuth } from "../context/auth-context";
-export const VideoOptions = ({ videoId }) => {
-  const { dispatchPlaylist } = usePlaylist();
+import { addToWatchLater } from "../utils/addToWatchLater";
+import { removeFromWatchLater } from "../utils/removeFromWatchLater";
+import { addToLikes } from "../utils/addToLikes";
+import { removeFromLikes } from "../utils/removeFromLikes";
+
+export const VideoOptions = ({ activeVideo }) => {
+  const { videoActionState, dispatchVideoAction } = useVideoAction();
   const { auth } = useAuth();
+
+  const isInWatchLaterList = (videoId) => {
+    return videoActionState.watchlater.some((item) => item._id === videoId);
+  };
+  const isInLikesList = (videoId) => {
+    return videoActionState.likes.some((item) => item._id === videoId);
+  };
+
   return (
     <div className="video-options-box flex-column">
       <RequiresAuth>
         <ul>
-          <li>
-            <i className="material-icons">watch_later</i>Save to watch later
+          <li
+            onClick={() => {
```

Comment:
If the onClick function block becomes too big/cluttered then you can separate it into a function.

Reply:
Thanks, will implement the same.